### PR TITLE
[gestalt] 53.1.1 April ( 28,2022)

### PR DESCRIPTION
**Pull request by zifahm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## gestalt ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gestalt@latest/))
was missing the following properties:
1. Fragment
1. jsx
1. jsxs
1. Collection
1. DeviceTypeProvider

<details>
<summary>as well as these 6 other properties...</summary>
<p>ExperimentProvider, I18nProvider, MasonryDefaultLayout, MasonryUniformRowLayout, ScrollFetch, useColorScheme</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 404561d9cdc885a805d42c96c0885b38b2f1f4f0
</p>

File: types/gestalt/index.d.ts [hunk 1]
```diff
@@ -392,10 +392,10 @@ export interface CheckboxProps {
     indeterminate?: boolean | undefined;
     label?: string | undefined;
     name?: string | undefined;
-
     onClick?: AbstractEventHandler<React.SyntheticEvent<HTMLInputElement>, { checked: boolean }> | undefined;
     size?: 'sm' | 'md' | undefined;
     subtext?: string | undefined;
+    labelDisplay?: 'visible' | 'hidden';
```
Comment (peterblazejewicz) [inline types/gestalt/index.d.ts:398]:
```suggestion
    labelDisplay?: 'visible' | 'hidden' | undefined;
```
as per `size`?

File: types/gestalt/index.d.ts [hunk 2]
```diff
@@ -1660,6 +1643,7 @@ export interface TextAreaProps {
     tags?: ReadonlyArray<React.ReactElement<TagProps, typeof Tag>> | undefined;
     value?: string | undefined;
     readonly?: boolean;
+    labelDisplay?: 'visible' | 'hidden';
```
Comment (peterblazejewicz) [inline types/gestalt/index.d.ts:1646]:
```suggestion
    labelDisplay?: 'visible' | 'hidden' | undefined;
```
can this is `undefined` in native code? (as per e.g. `type`)

File: types/gestalt/index.d.ts [hunk 3]
```diff
@@ -1706,6 +1690,7 @@ export interface TextFieldProps {
      */
     type?: 'date' | 'email' | 'password' | 'text' | 'url' | 'tel' | undefined;
     value?: string | undefined;
+    labelDisplay?: 'visible' | 'hidden';
```
Comment (peterblazejewicz) [inline types/gestalt/index.d.ts:1693]:
```suggestion
    labelDisplay?: 'visible' | 'hidden' | undefined;
```
ditto

**Comment by zifahm** [comment]
@peterblazejewicz  arnt we already saying ```label?:``` where "?" says its optional, why say undefined?

**Comment by peterblazejewicz** [comment]
@zifahm we are promoting better interop between TS libs, to support exact optional property types language feature intrduced in TS 4.4. That fully optional, but this is why you can see those added in here by DT mantainer. The support is optional and should be based on underlying module implementation (i.e. if it realy support passing 'undefined' values directly):
https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#exact-optional-property-types

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by zifahm** [comment]
Ready to merge

**Comment by peterblazejewicz** [comment]
Merged manually, assuming bot was on banking holiday

**Pull request merged**